analyzers/rspec/cs/S3900_c#.html

```diff
@@ -31,38 +27,58 @@ <h2>Compliant Solution</h2>
     private MyOtherClass other;
 
     public void Foo(MyOtherClass other)
-    {
-        this.other = other;
-    }
-
-    public void Bar(MyOtherClass other)
     {
         if (other != null)
         {
             this.other = other.Clone();
         }
     }
 
-    protected void FooBar(MyOtherClass other)
+    protected void Bar(MyOtherClass other)
     {
         if (other != null)
         {
             this.other = other.Clone();
         }
     }
+
+    public void Baz(MyOtherClass other)
+    {
+        ArgumentNullException.ThrowIfNull(other);
```

Comment:
We support this now. The test is in #7039 which didn't make it to master yet. We can keep the sample.